File: Midas/MIDAS.GBX/GBBusinessObjects/Doctor/Doctor.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MIDAS.GBX.BusinessObjects
{
    public class Doctor : GbObject
    {
        [Required]
        [JsonProperty("licenseNumber")]
        public string LicenseNumber { get; set; }

        [Required]
        [JsonProperty("wcbAuthorization")]
        public string WCBAuthorization { get; set; }

        [Required]
        [JsonProperty("wcbratingCode")]
        public string WcbRatingCode { get; set; }

        [Required]
        [JsonProperty("npi")]
        public string NPI { get; set; }

        
        [JsonProperty("taxType")]
        public GBEnums.TaxType TaxType { get; set; }

        [Required]
        [JsonProperty("title")]
        public string Title { get; set; }

        [Required]
        [JsonProperty("isCalendarPublic")]
        public bool IsCalendarPublic { get; set; }

        [Required]
        [JsonProperty("genderId")]
        public Byte GenderId { get; set; }

        [JsonProperty("user")]
        public User user { get; set; }

        [JsonProperty("doctorSpecialities")]
        public List<DoctorSpeciality> DoctorSpecialities { get; set; }

        [JsonProperty("doctorLocationSchedules")]
        public List<DoctorLocationSchedule> DoctorLocationSchedules { get; set; }

        [JsonProperty("doctorRoomTestMappings")]
        public List<DoctorRoomTestMapping> DoctorRoomTestMappings { get; set; }
    }

    public class MapDoctorToCompnay : GbObject
    {
        [JsonProperty("userName")]
        public string UserName { get; set; }
        [JsonProperty("companyID")]
        public int CompanyID { get; set; }
        [JsonProperty("currentUserId")]
        public int CurrentUserId { get; set; }

    }
    public class mDoctor : GbObject
    {
        [Required]
        [JsonProperty("licenseNumber")]
        public string LicenseNumber { get; set; }

        [Required]
        [JsonProperty("wcbAuthorization")]
        public string WCBAuthorization { get; set; }

        [JsonProperty("taxType")]
        public GBEnums.TaxType TaxType { get; set; }

        [Required]
        [JsonProperty("title")]
        public string Title { get; set; }

        [Required]
        [JsonProperty("isCalendarPublic")]
        public bool IsCalendarPublic { get; set; }

        public List<mDoctorSpeciality> mDoctorSpecialities { get; set; }
        public List<DoctorRoomTestMapping> mDoctorRoomTestMappings { get; set; }
        public List<DoctorLocationSchedule> DoctorLocationSchedules { get; set; }
    }
}

File: web-ui-with-angular-cli/src/app/medical-provider/users/models/doctor.ts
import { Record } from 'immutable';
import * as moment from 'moment';
import { User } from '../../../commons/models/user';
import { DoctorSpeciality } from './doctor-speciality';
import { DoctorLocationSchedule } from './doctor-location-schedule';

const DoctorRecord = Record({
    id: 0,
    licenseNumber: '',
    wcbAuthorization: '',
    wcbRatingCode: '',
    npi: '',
    taxType: '',
    title: '',
    userId: 0,
    user: null,
    genderId:1,
    doctorSpecialities: null,
    doctorLocationSchedules: null,
    isCalendarPublic: false,
    isDeleted: false,
    createByUserId: 0,
    updateByUserId: 0,
    createDate: null, //Moment
    updateDate: null //Moment
});

export class Doctor extends DoctorRecord {

    id: number;
    licenseNumber: string;
    wcbAuthorization: string;
    wcbRatingCode: string;
    npi: string;
    taxType: string;
    title: string;
    userId: number;
    user: User;
    genderId:number;
    doctorSpecialities: DoctorSpeciality[];
    doctorLocationSchedules: DoctorLocationSchedule[]; 
    isCalendarPublic: boolean;
    isDeleted: boolean;
    createByUserId: number;
    updateByUserId: number;
    createDate: moment.Moment;
    updateDate: moment.Moment;


    constructor(props) {
        super(props);
    }
}